### ISSUE TITLE
Trouble Extracting 7zip Archive in Multiple Volumes

### ISSUE THREAD

**Issue by functionreturnfunction** [cake-contrib/Cake.7zip#78]
Thanks very much for writing and sharing this!

I'm having an issue extracting a rather large archive in multiple volumes.  I know a lot of the processing is being done by the 7-Zip executable that comes with `7-Zip.CommandLine`, but I'm able to extract the same archive using `tools\7-Zip.CommandLine.18.1.0\tools\x64\7za.exe e "path\to\archive.7z.001" -o"path\to\destination\folder"` (the same executable that my cake script is using via Cake.7Zip).

I'm using the SevenZip alias as follows:

```c#
SevenZip(s => s
    .InExtractMode()
    .WithArchive(Path.Combine("path", "to", "archive.7z.001"))
    .WithArchiveType(SwitchArchiveType.SevenZip)
    .WithOutputDirectory(Path.Combine("path", "to", "destination")));
```

The result being:

```
ERROR: path\to\archive.7z.001
path\to\archive.7z.001
Open ERROR: Can not open the file as [7z] archive


ERRORS:
Unexpected end of archive
An error occurred when executing task 'Data-Import'.
Error: One or more errors occurred.
        7-Zip: Process returned an error (exit code 2).
```

Is there something I'm missing?  I'd find it very handy if I could somehow get the command builder to dump the command that would end up being executed to STDOUT so I can examine it.

**Comment by nils-a** [comment]
@functionreturnfunction sorry to hear that you're having a problem. I'll have a look on what the problem might be.

In the meantime - if you're running cake in diagnostic-mode (`--verbosity=Diagnostic` if you're using the bootstappers) the commandline that will be run ist printed out (along with many more infos) like so:

```
> build.ps1 --target=Test01 --verbosity=Diagnostic

Preparing to run build script...
Running build script...
[many lines of infos...]
========================================
Test01
========================================
Executing task: Test01
Executing: "D:/_dev/Cake.7zip/demo/tools/7-Zip.CommandLine.18.1.0/tools/x64/7za.exe" x -t7z -o"output/Test01" -y "output/volume.7z.001"
ERROR: output\volume.7z.001
output\volume.7z.001
Open ERROR: Can not open the file as [7z] archive


ERRORS:
Unexpected end of archive
An error occurred when executing task 'Test01'.
Error: System.AggregateException: One or more errors occurred. ---> Cake.Core.CakeException: 7-Zip: Process returned an error (exit code 2).
[man more lines...]
```

**Comment by nils-a** [comment]
@functionreturnfunction The problem seems to be `.WithArchiveType(SwitchArchiveType.SevenZip)` which translates to `-t7z` on the commandline. If I call that on a splitted volume I get the exact same error fom `7z.exe`

Running
```
SevenZip(s => s
    .InExtractMode()
    .WithArchive(Path.Combine("path", "to", "archive.7z.001"))
    .WithOutputDirectory(Path.Combine("path", "to", "destination")));
```
works as expected for me.

Can you confirm that?

**Comment by functionreturnfunction** [comment]
@nils-a The archive is split into 250 MB volumes, when I run my script with the `.WithArchiveType(...)` line commented out I get a single 250 MB file with the ".001" extension removed (the file I'm expecting from the archive is 5.7 GB).

It might bear mentioning at this point that the file is actually called `archive.bak.001` and contains a file called `archive.bak`.  I've also tried renaming the archive files to `archive.7z.nnn`, which resulted in: `Open ERROR: Can not open the file as [7z] archive`

**Comment by functionreturnfunction** [comment]
And just to be sure I've opened the volumes in my locally-installed 7zip instance to make sure I can still extract the file successfully, which i still can so I know the archive isn't corrupt.

**Comment by functionreturnfunction** [comment]
@nils-a Thanks for the suggestion to use `--verbosity=Diagnostic`, that's going to help me a lot when authoring cake scripts.

I noticed that the command being generated uses `x` instead of `e`, and the only time I've successfully been able to extract this archive using the executable was using `e`, so I added `.WithoutFullPathExtraction()` to my script but it still created a 250MB file.  However I tried the following commands on the command line and am somewhat baffled by the results:

```
C:\projects\my_project>"\projects\my_project\tools\7-Zip.CommandLine.18.1.0\tools\x64\7za.exe" e "\projects\other_project\Sql Backups\tmp\archive.bak.001" -o"\projects\other_project\Sql Backups\tmp\"

7-Zip (a) 18.01 (x64) : Copyright (c) 1999-2018 Igor Pavlov : 2018-01-28

Scanning the drive for archives:
1 file, 262144000 bytes (250 MiB)

Extracting archive: \projects\other_project\Sql Backups\tmp\archive.bak.001
--
Path = \projects\other_project\Sql Backups\tmp\archive.bak.001
Type = Split
Physical Size = 262144000
Volumes = 1
Total Physical Size = 262144000

Everything is Ok

Size:       262144000
Compressed: 262144000

C:\projects\my_project>tools\7-Zip.CommandLine.18.1.0\tools\x64\7za.exe e "..\other_project\Sql Backups\tmp\archive.bak.001" -o"..\other_project\Sql Backups\tmp\"

7-Zip (a) 18.01 (x64) : Copyright (c) 1999-2018 Igor Pavlov : 2018-01-28

Scanning the drive for archives:
1 file, 262144000 bytes (250 MiB)

Extracting archive: ..\other_project\Sql Backups\tmp\archive.bak.001
--
Path = ..\other_project\Sql Backups\tmp\archive.bak.001
Type = Split
Physical Size = 262144000
Volumes = 22
Total Physical Size = 5651318655
----
Path = archive.bak
Size = 5651318655
--
Path = archive.bak
Type = 7z
Physical Size = 5651318655
Headers Size = 146
Method = LZMA2:24
Solid = -
Blocks = 1


Would you like to replace the existing file:
  Path:     ..\other_project\Sql Backups\tmp\archive.bak
  Size:     262144000 bytes (250 MiB)
Modified: 2020-10-07 00:08:51
with the file from archive:
  Path:     archive.bak
  Size:     5786671104 bytes (5519 MiB)
  Modified: 2020-10-06 23:02:23
? (Y)es / (N)o / (A)lways / (S)kip all / A(u)to rename all / (Q)uit?
```

7za.exe seems to properly detect the file type and extract it when using a relative path, but not a full path from the root of my drive.  Very strange...

**Comment by nils-a** [comment]
Strange indeed. I can not confirm that observation (absolute vs. relative paths). 

```
❯ ."D:/_dev/Cake.7zip/demo/tools/7-Zip.CommandLine.18.1.0/tools/x64/7za.exe" x -o"output/Test01" -y "D:\_dev\Cake.7zip\demo\output\volume.7z.001"

7-Zip (a) 18.01 (x64) : Copyright (c) 1999-2018 Igor Pavlov : 2018-01-28

Scanning the drive for archives:
1 file, 52428800 bytes (50 MiB)

Extracting archive: D:\_dev\Cake.7zip\demo\output\volume.7z.001
--
Path = D:\_dev\Cake.7zip\demo\output\volume.7z.001
Type = Split
Physical Size = 52428800
Volumes = 6
Total Physical Size = 313785943
----
Path = volume.7z
Size = 313785943
--
Path = volume.7z
Type = 7z
Physical Size = 313785943
Headers Size = 126882
Method = LZMA2:24 BCJ
Solid = +
Blocks = 2

Everything is Ok

Folders: 1362
Files: 6411
Size:       1317148826
Compressed: 313785943

❯ ."tools/7-Zip.CommandLine.18.1.0/tools/x64/7za.exe" x -o"output/Test01" -y "D:\_dev\Cake.7zip\demo\output\volume.7z.001"

7-Zip (a) 18.01 (x64) : Copyright (c) 1999-2018 Igor Pavlov : 2018-01-28

Scanning the drive for archives:
1 file, 52428800 bytes (50 MiB)

Extracting archive: D:\_dev\Cake.7zip\demo\output\volume.7z.001
--
Path = D:\_dev\Cake.7zip\demo\output\volume.7z.001
Type = Split
Physical Size = 52428800
Volumes = 6
Total Physical Size = 313785943
----
Path = volume.7z
Size = 313785943
--
Path = volume.7z
Type = 7z
Physical Size = 313785943
Headers Size = 126882
Method = LZMA2:24 BCJ
Solid = +
Blocks = 2

Everything is Ok

Folders: 1362
Files: 6411
Size:       1317148826
Compressed: 313785943

❯
```

Could you please, on the commandline, test whether specitying the type as `-t7z.split` (or `-t"7z.split"`) make any difference. (The ".split"-postfix for the archive-type is not documented as such, but there is one example using that syntax and it worked for me...)

**Comment by nils-a** [comment]
@functionreturnfunction I have released [version 0.7.1](https://www.nuget.org/packages/Cake.7zip/0.7.1). With that you can call 7zip like this:
```
SevenZip(s => s
    .InExtractMode()
    .WithArchive(Path.Combine("path", "to", "archive.7z.001"))
    .WithArchiveType(SwitchArchiveType.SevenZip.Volumes())
    .WithOutputDirectory(Path.Combine("path", "to", "destination")));
```

which will translate to `... -t7z.split ...` - using this extract of the archive *should* work.

**Comment by functionreturnfunction** [comment]
@nils-a Sorry if I've wasted your time here, but I grabbed a fresh set of archives this morning and now I can't get extraction to _fail_.  Something must have been up with yesterday's files, when I tried `-t7z.split` it again failed to recognize the archive as 7z.  Thank you very much for looking into this with me.
